app/models/miq_cockpit_ws_worker.rb

```diff
@@ -0,0 +1,118 @@
+require 'miq_apache'
+require 'runcmd'
+require 'linux_admin'
+
+class MiqCockpitWsWorker < MiqWorker
+  APACHE_CONF_FILE = '/etc/httpd/conf.d/manageiq-cockpit.conf'
+  BINDING_ADDRESS = ENV['BINDING_ADDRESS'] || (Rails.env.production? ? nil : "0.0.0.0")
+
+  self.required_roles = ['cockpit_ws']
+  self.maximum_workers_count = 1
+
+  def friendly_name
+    @friendly_name ||= "Cockpit Worker"
+  end
+
+  def self.can_activate_cockpit? (opts = nil)
+    if self.has_required_role?
+      if (!opts)
```

Comment:
you can use
`opts ||= MiqCockpitWsWorker.worker_settings()`
instead of this `if` clause. This is applicable several times in this patch.